src/fstext/trivial-factor-weight.h

```diff
@@ -307,19 +292,18 @@ class TrivialFactorWeightFstImpl
 
   typedef unordered_map<Element, StateId, ElementKey, ElementEqual> ElementMap;
 
-  const Fst<A> *fst_;
+  std::unique_ptr<const Fst<A>> fst_;
   float delta_;
```

Comment:
I see bellow this that you initialize unique_ptr using make_shared. Is that a correct way how to handle this? I know that make_unique exists only since C++14, but I'm not sure about interactions between assingments of unique_ptr and shared_ptr